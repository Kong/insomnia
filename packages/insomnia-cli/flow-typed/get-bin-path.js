// @flow

declare module 'get-bin-path' {
  declare module.exports: *;
}
