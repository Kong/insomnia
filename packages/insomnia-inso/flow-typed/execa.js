// @flow

declare module 'execa' {
  declare module.exports: *;
}
