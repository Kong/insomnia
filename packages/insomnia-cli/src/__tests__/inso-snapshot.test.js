// @flow
import execa from 'execa';
import { getBinPathSync } from 'get-bin-path';

// MAKE SURE YOU BUILD THE PROJECT BEFORE RUNNING THESE TESTS.
// These tests use the executable /bin/inso, which relies on /dist.

describe('Snapshot for', () => {
  it.each(['-h', '--help', 'help', 'generate -h', 'generate config -h'])(
    '"inso %s"',
    async args => {
      const { stdout } = await execa.node(getBinPathSync(), args.split(' '));
      expect(stdout).toMatchSnapshot();
    },
  );
});
