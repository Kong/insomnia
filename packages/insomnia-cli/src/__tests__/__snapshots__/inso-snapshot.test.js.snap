// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot for "inso --help" 1`] = `
"Usage: inso [options] [command]

A CLI for Insomnia!

Options:
  -v, --version       output the version number
  --workingDir <dir>  Working directory
  -h, --help          display help for command

Commands:
  generate            Code generation utilities
  help [command]      display help for command"
`;

exports[`Snapshot for "inso -h" 1`] = `
"Usage: inso [options] [command]

A CLI for Insomnia!

Options:
  -v, --version       output the version number
  --workingDir <dir>  Working directory
  -h, --help          display help for command

Commands:
  generate            Code generation utilities
  help [command]      display help for command"
`;

exports[`Snapshot for "inso generate -h" 1`] = `
"Usage: inso generate [options] [command]

Code generation utilities

Options:
  -h, --help                     display help for command

Commands:
  config [options] <identifier>  Generate configuration from an api spec
  help [command]                 display help for command"
`;

exports[`Snapshot for "inso generate config -h" 1`] = `
"Usage: inso generate config [options] <identifier>

Generate configuration from an api spec

Options:
  -t, --type <value>   the type of configuration to generate, options are
                       [kubernetes, declarative]
  -o, --output <path>  the output path
  -h, --help           display help for command"
`;

exports[`Snapshot for "inso help" 1`] = `
"Usage: inso [options] [command]

A CLI for Insomnia!

Options:
  -v, --version       output the version number
  --workingDir <dir>  Working directory
  -h, --help          display help for command

Commands:
  generate            Code generation utilities
  help [command]      display help for command"
`;
